File: testdata/typescript/valid.ts
/**
 * Represents a user in the system.
 */
interface User {
    id: number;
    name: string;
    email: string;
}

/**
 * Configuration options for the application.
 */
type Config = {
    debug: boolean;
    timeout: number;
};

/**
 * Greeting service for handling user greetings.
 */
class GreetingService {
    private prefix: string;

    /**
     * Creates a new GreetingService.
     * @param prefix The greeting prefix
     */
    constructor(prefix: string) {
        this.prefix = prefix;
    }

    /**
     * Greets a user.
     * @param user The user to greet
     * @returns The greeting message
     */
    greet(user: User): string {
        return `${this.prefix}, ${user.name}!`;
    }
}

/**
 * Formats a user for display.
 * @param user The user to format
 * @returns Formatted string
 */
function formatUser(user: User): string {
    return `${user.name} <${user.email}>`;
}

const DEFAULT_TIMEOUT = 5000;

export { User, Config, GreetingService, formatUser, DEFAULT_TIMEOUT };

File: testdata/go/valid.go
package main

import "fmt"

// Server represents the main application server.
type Server struct {
	Name string
	Port int
}

// NewServer creates a new Server instance.
func NewServer(name string, port int) *Server {
	return &Server{
		Name: name,
		Port: port,
	}
}

// Start starts the server.
func (s *Server) Start() error {
	fmt.Printf("Starting server %s on port %d\n", s.Name, s.Port)
	return nil
}

// Config holds application configuration.
type Config struct {
	Debug   bool
	Timeout int
}

const (
	// DefaultPort is the default server port.
	DefaultPort = 8080
)

var (
	// Version is the application version.
	Version = "1.0.0"
)

func main() {
	srv := NewServer("main", DefaultPort)
	srv.Start()
}
